specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json

```diff
@@ -0,0 +1,488 @@
+          "readOnly": true
+        }
+      },
+      "description": "The builder properties.",
+      "required": [
+        "environmentId"
+      ]
+    },
+    "BuilderResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/BuilderProperties",
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-client-flatten": true,
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v3/managedidentity.json#/definitions/ManagedServiceIdentity",
+          "description": "The managed service identities assigned to this resource."
+        }
+      },
+      "description": "Information about the SourceToCloud builder resource.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ]
+    },
+    "BuilderCollection": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BuilderResource"
+          },
+          "description": "The BuilderResource items on this page"
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "description": "The response of a BuilderResource list operation.",
+      "required": [
+        "value"
+      ]
+    },
+    "BuilderResourceUpdate": {
+      "type": "object",
+      "properties": {
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v3/managedidentity.json#/definitions/ManagedServiceIdentity",
+          "description": "The managed service identities assigned to this resource."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags."
+        },
+        "properties": {
+          "$ref": "#/definitions/BuilderResourceUpdateProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "description": "The type used for update operations of the BuilderResource."
+    },
+    "BuilderResourceUpdateProperties": {
+      "type": "object",
+      "properties": {
+        "environmentId": {
+          "type": "string",
+          "description": "Resource ID of the container apps environment that the builder is associated with."
+        }
+      },
+      "description": "The updatable properties of the BuilderResource."
+    },
+    "ProvisioningState": {
+      "type": "string",
+      "description": "Resource instance provisioning state.",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled",
+        "InProgress",
```

Comment:
>InProgress [](http://example.com/codeflow?start=9&length=10)

When is this state used? Consider something more specific (e.g. Creating, Initializing, Updating, or similar).

Reply:
updated to be creating

Reply:
Ideally, you could distinguish Creating and Updating and have both since PUT can be either a create or update. Just Creating is also OK, not blocking ARM signoff.

Reply:
That makes sense. I'll add `updating` as well. Just want to clarify one thing quickly.

The [RPC ](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#provisioningstate-property) mentions that there are the three terminal states. It sounds like we can have a scenario (like when we are `updating`), when the resource is able to transition out of the terminal state (of `succeeded`) and into another non-terminal state (`updating`), and then eventually back into the terminal state. 

Is this correct? I was originally interpreting a terminal state as a "final" state from which the resource would not be able to transition, but sounds like that is not right. Is it just terminal for each async operation, and not the resource as a whole?

Reply:
Correct. I think terminal here means it won't be changed by the RP, only in response to a client request.